### PR TITLE
Replace userId with userSession->UID

### PR DESCRIPTION
**Pull request by jotoeri**
Cleaning a Code-Todo, using the userSession instead of userId.
Old code was a bit better to read imo. @skjnldsv ;)

### CODE REVIEW

File: lib/Controller/ApiController.php [hunk 1]
```diff
@@ -207,7 +205,7 @@ public function updateForm(int $id, array $keyValuePairs): Http\JSONResponse {
 			return new Http\JSONResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
-		if ($form->getOwnerId() !== $this->userId) {
+		if ($form->getOwnerId() !== $this->userSession->getUser()->getUID()) {
```
Comment (skjnldsv) [inline lib/Controller/ApiController.php:208]: What if not logged in? :thinking:

File: lib/Controller/ApiController.php [hunk 2]
```diff
@@ -98,7 +97,6 @@ public function __construct(string $appName,
 								FormsService $formsService) {
 		parent::__construct($appName, $request);
 		$this->appName = $appName;
-		$this->userId = $userId;
```
Comment (jancborchardt) [inline lib/Controller/ApiController.php:101]: Just reinstate that, but as 
`$this->userSessionId = userSession->getUser()->getUID();`
to make it easier to read again? :)

File: lib/Controller/ApiController.php (hunk 1 above)
Comment (jotoeri) [reply lib/Controller/ApiController.php:208]: :D You tell me - userId was, what Roeland wrote here - you wanted to change to userSession, as you said it's cleaner... What does userSession return, if not logged in? These comparisons should just return true to return the forbidden-status within the if...

Comment (skjnldsv) [reply lib/Controller/ApiController.php:208]: getUser returns null I guess?

Comment (jotoeri) [reply lib/Controller/ApiController.php:208]: Yep - So it should work, as it has to...
grafik

grafik

Comment (kesselb) [reply lib/Controller/ApiController.php:208]: Question: The user must be logged in to use any of the ApiController methods?

Comment (kesselb) [reply lib/Controller/ApiController.php:208]: https://github.com/nextcloud/mail/blob/5f744699ca36aefa919668cb717ae15c045be89a/lib/Controller/AliasesController.php#L41-L50

That ($currentUser) might be a bit cleaner and $this->currentUser->getUID() is easier to read.

Comment (jotoeri) [reply lib/Controller/ApiController.php]: > Question: The user must be logged in to use any of the ApiController methods?

Mostly, yes. Currently only the form-owner is allowed to see responses & edit forms. Plans to extend this do exist. `insertSubmission()` is also usable without beeing logged in, but uses the currentUser for storing userId.

> That ($currentUser) might be a bit cleaner and $this->currentUser->getUID() is easier to read.

Hm, yes somehow similar to what Jan said. Did that now, and even changed on FormsService for similar usage.

**Comment by jotoeri** [comment]
Just pushed again for checks to run.

**Comment by jotoeri** [comment]
Ok, @jancborchardt so i think we can go here? It's just a bit of standardized code-replacement. I just systematically went through all functions, functionality seems te be working everywhere.

**Comment by jancborchardt** [comment]
> Ok, @jancborchardt so i think we can go here? It's just a bit of standardized code-replacement. I just systematically went through all functions, functionality seems te be working everywhere.

Sounds good to me. Can you check out the file conflict? Then afterwards we can merge. :)

**Comment by jotoeri** [comment]
:D that was the logger now.^^

**Pull request merged**